compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -157,15 +157,8 @@ object TypeOps:
         tp.derivedAlias(simplify(tp.alias, theMap))
       case AndType(l, r) if !ctx.mode.is(Mode.Type) =>
         simplify(l, theMap) & simplify(r, theMap)
```

Comment:
Should this be calling into `TypeComparer.glb` It just seems odd to be altering the Or case, but not the And case.

Reply:
`&` and glb are in fact the same. It's just for `lub` that we need the additional `isSoft` parameter.